### PR TITLE
VOSM website: 2.0

### PR DESCRIPTION
**Pull request by frvos**
# VOSM 2.0 Major Refactoring
- Move to MongoDB and introduce these domain instances instead of relying on tracking Stripe's sessions data:
  - Reservation
  - Registration (representing someone able to register on behalf of participants)
  - Participants
  - Meeting (new concept)
  this allow us to have different meeting rules each year/time

## Creating a Meeting
This change also allows us to have always up-to-date meeting rules. For the first time after deployment, you can create a meeting with the following cURL command:

```bash
curl --request POST \
  --url http://localhost:3001/api/create-meeting \
  --header 'Content-Type: application/json' \
  --data '{
    "title": "2024 VOSM",
    "description": "Description goes here",
    "maxParticipants": 200
}'
```

## Environtment Variables
```bash
MONGO_URI=mongodb+srv://vosm:<password>@cluster0.njmubqo.mongodb.net
STRIPE_SECRET_KEY=sk_test_***
NEXT_PUBLIC_STRIPE_PUBLISHABLE_KEY=pk_test_***
WEBHOOK_SECRET=whsec_***
```

### CODE REVIEW

**Review by phil-evoiq — APPROVED** [review]
LGTM

**Review by philarmour — APPROVED** [review]
LGTM

File: infra/database/VosmFaunaDatabaseService.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
+import { query as q } from 'faunadb';
```
Comment (frvos) [inline infra/database/VosmFaunaDatabaseService.ts:1]: deprecated

File: infra/StripeInteractor.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import queryString from 'query-string';
```
Comment (frvos) [inline infra/StripeInteractor.ts:1]: all API call to stripe goes here

File: infra/database/mongo-database-service/MongoDatabaseService.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import mongoose, { Mongoose } from 'mongoose';
```
Comment (frvos) [inline infra/database/mongo-database-service/MongoDatabaseService.ts:1]: I believe this is the biggest change, @philarmour. Another significant update involves create-stripe-participants-sessions.ts, in addition to cleaning up the webhook. The rest are just routine tasks.

**Pull request merged**